pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -761,6 +773,73 @@ func (r *ReconcileClusterDeployment) startNewProvision(
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileClusterDeployment) copyInstallLogSecret(destNamespace string, extraEnvVars []corev1.EnvVar) error {
+	hiveNS := controllerutils.GetHiveNamespace()
+
+	srcSecretName, foundSrc := os.LookupEnv(constants.InstallLogsAWSCredentialsSecretRefEnvVar)
+	if !foundSrc {
+		// If the src secret reference wasn't found, then don't attempt to copy the secret.
+		return nil
+	}
+
+	foundDest := false
+	var destSecretName string
+	for _, envVar := range extraEnvVars {
+		if envVar.Name == constants.InstallLogsAWSCredentialsSecretRefEnvVar {
+			destSecretName = envVar.Value
+			foundDest = true
+		}
+	}
+
+	if !foundDest {
+		// If the dest secret reference wasn't found, then don't attempt to copy the secret.
+		return nil
+	}
+
+	src := types.NamespacedName{Name: srcSecretName, Namespace: hiveNS}
+	dest := types.NamespacedName{Name: destSecretName, Namespace: destNamespace}
+	return controllerutils.CopySecret(r, src, dest)
+}
+
+func getInstallLogEnvVars(secretPrefix string) []corev1.EnvVar {
+	extraEnvVars := []corev1.EnvVar{}
+
+	cloudProvider, found := os.LookupEnv(constants.InstallLogsUploadProviderEnvVar)
+	if !found {
+		return extraEnvVars
+	}
+
+	if cloudProvider == constants.InstallLogsUploadProviderAWS {
+		secretName, foundSrc := os.LookupEnv(constants.InstallLogsAWSCredentialsSecretRefEnvVar)
+		if foundSrc {
+			extraEnvVars = append(extraEnvVars, corev1.EnvVar{
+				Name:  constants.InstallLogsAWSCredentialsSecretRefEnvVar,
+				Value: secretPrefix + "-" + secretName,
+			})
+		}
+
+		extraEnvVars = addEnvVarIfFound(constants.InstallLogsAWSRegionEnvVar, extraEnvVars)
```

Comment:
Would it make adding future providers cleaner if we stored the `installLogsUploadProvider` in the `extraEnvVars` (so that we can `switch` on the value of the provider?

Reply:
Done.